src/promptflow-evals/promptflow/evals/evaluators/_content_safety/_self_harm.py

```diff
@@ -1,15 +1,24 @@
 # ---------------------------------------------------------
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # ---------------------------------------------------------
-try:
-    from .common.constants import EvaluationMetrics
-    from ._content_safety_sub_evaluator_base import ContentSafetySubEvaluatorBase
-except ImportError:
-    from common.constants import EvaluationMetrics
-    from _content_safety_sub_evaluator_base import ContentSafetySubEvaluatorBase
+from promptflow._utils.async_utils import async_run_allowing_running_loop
+from promptflow.evals.evaluators._content_safety.common import ContentSafetyEvaluatorBase
+from promptflow.evals.evaluators._content_safety.common.constants import EvaluationMetrics
 
 
-class SelfHarmEvaluator(ContentSafetySubEvaluatorBase):
+class _AsyncSelfHarmEvaluator(ContentSafetyEvaluatorBase):
+    def __init__(self, project_scope: dict, credential=None):
+        super().__init__(
+            metric=EvaluationMetrics.SELF_HARM,
+            project_scope=project_scope,
+            credential=credential,
+        )
+
+    async def __call__(self, *, question: str, answer: str, **kwargs):
```

Comment:
Do we need to redefine the __call__ here?

Reply:
If I understand the question correctly, the answer is yes. We will need an async version of the `\_\_call\_\_` method to work with the async code path in the batch engine.